listener/udp.go

```diff
@@ -0,0 +1,109 @@
+// Package listener handles the ingestion for polymur
+package listener
+
+import (
+	"log"
+	"net"
+	"time"
+
+	"github.com/chrissnell/polymur/statstracker"
+)
+
+// UDPListenerConfig holds configuration for the UDP listener
+type UDPListenerConfig struct {
+	IncomingQueue chan []*string
+	FlushTimeout  int
+	FlushSize     int
+	Stats         *statstracker.Stats
+	IP            string
+	Port          int
+	Zone          string
+}
+
+// UDPListener Listens for messages.
+func UDPListener(config *UDPListenerConfig) {
+	log.Printf("Metrics listener started: [UDP] %s:%d\n", config.IP, config.Port)
+	server, err := net.ListenUDP(
+		"udp",
+		&net.UDPAddr{IP: net.ParseIP(config.IP), Port: config.Port},
+	)
+	if err != nil {
+		log.Fatalf("Listener error: %s\n", err)
+	}
+
+	messages := make(chan string, 128)
+	go udpMessageBatcher(messages, config)
```

Comment:
I need some more time to figure out what's going on here.  It's a funky pattern that you're doing here with the anonymous goroutine to read the UDP.  Let's talk about it tomorrow.

In the meantime, I want to show you a pattern that I learned recently that's very common in well written Go.  Whenever I need to have a consumer that's consuming and processing channel messages, I create it and launch it like this:

https://github.com/chrissnell/crabby/blob/c664934515934e5a49c292f52616148a5c7d2ec3/storage_graphite.go#L30-L61